Analysis/src/Frontend.cpp

```diff
@@ -697,6 +685,31 @@ ScopePtr Frontend::getModuleEnvironment(const SourceModule& module, const Config
         }
     }
 
+    if (!config.globalTypePaths.empty())
+    {
+        unfreeze(typeChecker.globalTypes);
+
+        result = std::make_shared<Scope>(result);
+        for (const std::string& path : config.globalTypePaths)
+        {
+            ModulePtr module = moduleResolver.getModule(path);
+            if (!module)
+            {
+                auto sourceCode = fileResolver->readSource(path);
+                if (!sourceCode)
+                    continue;
+                SourceModule sourceModule = parse(path, sourceCode->source, {});
+                module = typeChecker.check(sourceModule, Mode::Strict);
+                if (!module)
+                    continue;
+            }
+
+            loadModuleIntoScope(typeChecker, module, result, path);
+        }
+
+        freeze(typeChecker.globalTypes);
+    }
```

Comment:
Would this not cause new types to be created for the scope every time `getModuleEnvironment` is called (which is done each time `Frontend::check` is called)?

Reply:
It should only be checked the first time around, that's why we check if the module exists. If it doesn't then we'll check it. However, we should probably also check if it's dirty.

Reply:
This should now be resolved, as there aren't any copies taking place anymore.